spatialdata/_core/_spatialdata.py

```diff
@@ -430,40 +450,59 @@ def add_image(
             See https://zarr.readthedocs.io/en/stable/api/storage.html for more details.
         overwrite
             If True, overwrite the element if it already exists.
-        _add_in_memory
+        _called_from_write
             Internal flag, to differentiate between an element added by the user and an element saved to disk by
             write method.
```

Comment:
for this behaviour in general I would like to ask @ivirshup to comment

Reply:
I removed this behavior, I realized it's bad practice because the user may trigger untested behaviors. Now the code used from `write()` is inside `write()`, and the `add_image()`, `add_labels()`, ... functions are not called from `write()`.